### PR TITLE
Allow other schemas, database and groups on tree view

### LINKED ISSUE #71: Schemas are not supported (postgresql)
> This is a guideline to help us to understand SQLTools Issues. 
> 
> You don't need to stick to this, but please try to guide us to reproduce the errors or understand your feature requests.

### Issue Type

* [ ] Feature Request
* [x ] Bugfix/Errors
* [ ] Questions
* [ ] Other

### Prerequisites (For bugfixes)

* [x ] Are you running the latest version?
* [ ] Did you check the logs?
* [ ] Did you check the Setup?

### Description

[Description of the bug or feature]
I have a database where all tables belons to different schemas. This extension lists all tables as if they are in database (root schema)

### Steps to Reproduce (For bugfixes)

1. [First Step]
2. [Second Step]
3. [and so on...]

**Expected behavior:** [What you expected to happen]

**Actual behavior:** [What actually happened]

### Versions

- *Version*: vX.Y.Z
- *OS*: ...
- *SGDB*: MySQL, PostgresSQL...

You can get this information from executing `SQLTools: Version` from VSCode `Quick Open`.

### PR DESCRIPTION
**Pull request by mtxr**
This PR will close #71.

### CODE REVIEW

File: packages/core/dialect/pgsql/queries.ts [hunk 1]
```diff
@@ -15,7 +15,13 @@ SELECT
...
   C.COLUMN_DEFAULT AS defaultValue,
   C.IS_NULLABLE AS isNullable,
-  TC.constraint_type AS keytype
+  TC.constraint_type AS keytype,
+  C.TABLE_CATALOG || '/' || C.TABLE_SCHEMA  || '/' || (
+    CASE
+      WHEN T.TABLE_TYPE = 'VIEW' THEN 'views'
+      ELSE 'tables'
+    END
+  ) || '/' || C.TABLE_name || '/' || C.COLUMN_NAME AS tree
```
Comment (mtxr) [inline packages/core/dialect/pgsql/queries.ts]: @mickeypearce can you do the same change on Oracle queries?

File: packages/core/dialect/sqlite/index.ts [hunk 2]
```diff
@@ -117,6 +118,7 @@ export default class SQLite extends GenericDialect<SQLiteLib.Database> implement
           size: null,
           isPk: Boolean(obj.pk),
           isFk: !!fksMap[obj.name],
+          tree: obj.tree,
```
Comment (mtxr) [inline packages/core/dialect/sqlite/index.ts]: Also later you should pass the pro here, and..

File: packages/plugins/connection-manager/explorer/index.ts [hunk 3]
```diff
@@ -108,34 +104,88 @@ export class ConnectionExplorer implements TreeDataProvider<SidebarDatabaseItem>
...
+    });
+    return tree.tree[created.pop()];
+  }
+
+  private insertTables(connId: string, dialect: DatabaseDialect, tables: DatabaseInterface.Table[]) {
+    switch (dialect) {
+      case DatabaseDialect.PostgreSQL:
+      case DatabaseDialect.SQLite:
+      case DatabaseDialect.MySQL:
+        tables.sort((a, b) => a.name.localeCompare(b.name)).forEach((item) => {
```
Comment (mtxr) [inline packages/plugins/connection-manager/explorer/index.ts]: Add the dialect condition here

File: packages/plugins/connection-manager/explorer/index.ts [hunk 4]
```diff
@@ -108,34 +104,88 @@ export class ConnectionExplorer implements TreeDataProvider<SidebarDatabaseItem>
...
+  }
+
+  private insertColumns(connId: string, dialect: DatabaseDialect, columns: DatabaseInterface.TableColumn[]) {
+    if (ConfigManager.sortColumns && ConfigManager.sortColumns === 'name') {
+      columns = columns.sort((a, b) => a.columnName.localeCompare(b.columnName));
+    } else if (ConfigManager.sortColumns && ConfigManager.sortColumns === 'ordinalnumber') { /* it's already sorted by position */}
+    switch (dialect) {
+      case DatabaseDialect.PostgreSQL:
+      case DatabaseDialect.SQLite:
+      case DatabaseDialect.MySQL:
```
Comment (mtxr) [inline packages/plugins/connection-manager/explorer/index.ts]: and here

**Comment by mtxr** [comment]
@mickeypearce please take a look at this comments.

After this changes it should work for oracle as well.

Sorry for this massive PR, but we had some valuable improvements here.

File: packages/plugins/connection-manager/extension.ts [hunk 5]
```diff
@@ -195,9 +195,15 @@ export default class ConnectionManagerPlugin implements SQLTools.ExtensionPlugin
...
   // internal utils
-  private async _getTableName(node?: SidebarTable | SidebarView): Promise<string> {
+  private async _getTableName(node?: SidebarTableOrView): Promise<string> {
     if (node && node.value) {
       await this._setConnection(node.conn as ConnectionInterface);
+      switch(node.conn.dialect) {
+        case DatabaseDialect.PostgreSQL:
+          return [node.table.tableDatabase, node.table.tableSchema, node.table.name].join('.');
+        case DatabaseDialect.MySQL:
+          return [node.table.tableSchema, node.table.name].join('.');
```
Comment (mtxr) [inline packages/plugins/connection-manager/extension.ts]: Also here, to make it query for the exact table name @jpincas was talking about.

**Comment by mickeypearce** [comment]
Ok, I'm doing it tomorrow. Looks nice.

**Comment by mtxr** [comment]
Ill close this PR. Im having some merging issues.

I'm opening a new one, ok?

**Comment by mickeypearce** [comment]
@mtxr, '@sqltools/formatter/src/sqlFormatter'; is missing from dependency?

**Comment by mickeypearce** [comment]
Aha, I see it is a git submodule...

**Pull request closed without merging**